Source/Classes/PINRemoteImageManager.m

```diff
@@ -97,6 +97,32 @@ BOOL PINRemoteImageManagerSubclassOverridesSelector(Class subclass, SEL selector
 NSString * const PINRemoteImageCacheKeyResumePrefix = @"R-";
 typedef void (^PINRemoteImageManagerDataCompletion)(NSData *data, NSURLResponse *response, NSError *error);
 
+@interface PINRemoteImageWeakContainer : NSObject
+
+@property (nonatomic, readonly, weak) id target;
+
++ (PINRemoteImageWeakContainer *)weakContainerWithTarget:(id)target;
+- (instancetype)initWithTarget:(id)target;
+
+@end
+
+@implementation PINRemoteImageWeakContainer
```

Comment:
I think this can have a better name.

Reply:
We could name this `PINRemoteImageWeakTask` now that it always holds `PINRemoteImageTask` objects.